### ISSUE TITLE
Logging and debugging

### ISSUE THREAD

**Issue by digitalhuman** [diaspora/diaspora#5921]
Hi there,

I'm Victor, PodAdmin of https://pod.readme.is

Something is bothering me for some time and that is the logging of Diaspora. I personally think its wrongly implemented and most important it's a risk factor, a privacy/security issue, and its useless for debugging purposes because it doesn't log any useful when things are going wrong.

The other part is that PodAdmins are not able to enable and disable logging and it's also not possible to choose a level of logging. (As far as I know).

I hope you guess will fix this in one of the next releases because I personally think this should be solved very soon. Diaspora stands for something right?..... this doesn't fit in that thought.

**Comment by Flaburgan** [comment]
Hi there. First of all, you are talking about production.log right? I agree about the level, to be able to set it on error would be interesting. Apart from that, what other improvements would you need exactly?

**Comment by digitalhuman** [comment]
Yes the production log. Well first of all the log level should be a setting. From 0 to 4 or whatever. Also the type of logging should be a setting. (error, debug, notice, whatever name)

To refer to my other 'issue' https://github.com/diaspora/diaspora/issues/5920 If i was able to for example set it to: debug, and errors only I would have been able to fix this issue myself.

This also will improve the performance of Diaspora* as a whole. Less logging, less IO, better performance, but most important thing. The less logging, the less privacy/security issues which should be no1 priority.

**Comment by jhass** [comment]
Someone would need to go through  the current log statements and decide which level they should have. The other open question is what the default level should be. I also don't follow what you mean with having a log level 0-4 and having a log level error to debug, that's the same, no?

**Comment by digitalhuman** [comment]
Well that is indeed the choice. To start with it should be great to at least be able to turn it on or off. The next step could be implementing the options: error, debug or notice OR level 0-3. So either one of the 2 options.
